lib/error/error.ex

```diff
@@ -1,729 +1,590 @@
-
-      ### Example
-        def exception(:not_integer) do
-          ...
-        end
-
-    Raises an exception error when the error is produced
+    Create a `XDR.Error.Int` exception with the message of the `error_type` passed.
     """
     def exception(:not_integer) do
-      msg = "The value which you try to encode is not an integer"
-      %XDR.Error.Int{message: msg}
+      "The value which you try to encode is not an integer"
+      |> new()
     end
 
     def exception(:not_binary) do
-      msg =
-        "The value which you try to decode must be a binary value, for example: <<0, 0, 0, 2>>"
-
-      %XDR.Error.Int{message: msg}
+      "The value which you try to decode must be a binary value, for example: <<0, 0, 0, 2>>"
+      |> new()
     end
 
     def exception(:exceed_upper_limit) do
-      msg =
-        "The integer which you try to encode exceed the upper limit of an integer, the value must be less than 2_147_483_647"
-
-      %XDR.Error.Int{message: msg}
+      "The integer which you try to encode exceed the upper limit of an integer, the value must be less than 2_147_483_647"
+      |> new()
     end
 
     def exception(:exceed_lower_limit) do
-      msg =
-        "The integer which you try to encode exceed the lower limit of an integer, the value must be more than -2_147_483_648"
-
-      %XDR.Error.Int{message: msg}
+      "The integer which you try to encode exceed the lower limit of an integer, the value must be more than -2_147_483_648"
+      |> new()
     end
+
+    @spec new(msg :: String.t()) :: struct()
+    defp new(msg), do: %XDR.Error.Int{message: msg}
   end
 
   defmodule UInt do
     @moduledoc """
-    This module is in charge of containing the errors that may be raised by the XDR.UInt module
+    This module contains the definition of `XDR.Error.UInt` exception that may be raised by the `XDR.UInt` module.
     """
+
     defexception [:message]
 
     @impl true
     @doc """
-    This function is in charge of raise the errors that can be returned by XDR.UInt module
-      ## Parameters:
-        the function only have one parameter that is an atom which represents the type of error returned by the module
-
-      ### Example
-        def exception(:not_list) do
-          ...
-        end
-
-    Raises an exception error when the error is produced
+    Create a `XDR.Error.UInt` exception with the message of the `error_type` passed.
     """
     def exception(:not_integer) do
-      msg = "The value which you try to encode is not an integer"
-      %XDR.Error.UInt{message: msg}
+      "The value which you try to encode is not an integer"
+      |> new()
     end
 
     def exception(:not_binary) do
-      msg =
-        "The value which you try to decode must be a binary value, for example: <<0, 0, 0, 2>>"
-
-      %XDR.Error.UInt{message: msg}
+      "The value which you try to decode must be a binary value, for example: <<0, 0, 0, 2>>"
```

Comment:
@marioyustikommit As a tip when your pipe operator has just one step the recommendation is to write the function like:  
````
function(param)
````

Reply:
Thank you Juan, it is done!